src/main/scala/com/rbmhtechnology/calliope/serializer/kafka/PayloadFormatSerializer.scala

```diff
@@ -48,22 +51,46 @@ class PayloadFormatSerializer[A <: AnyRef] private(serializer: PayloadSerializer
 }
 
 object PayloadFormatDeserializer {
-  def apply[A](payloadMapper: AnyRef => A)(implicit system: ActorSystem): PayloadFormatDeserializer[A] =
-    new PayloadFormatDeserializer[A](DelegatingStringManifestPayloadSerializer(system), payloadMapper)
 
   def apply()(implicit system: ActorSystem): PayloadFormatDeserializer[AnyRef] =
     apply[AnyRef](identity)
 
-  def create[A](payloadMapper: JFunction[AnyRef, A], system: ActorSystem): PayloadFormatDeserializer[A] =
-    apply(payloadMapper.apply)(system)
+  def apply[A](payloadMapper: AnyRef => A)(implicit system: ActorSystem): PayloadFormatDeserializer[A] =
+    instance(payloadMapper, err => throw err)
+
+  def option[A](payloadMapper: PartialFunction[AnyRef, A])(implicit system: ActorSystem): PayloadFormatDeserializer[Option[A]] =
```

Comment:
@krasserm How would you generate only elements that match the `PartialFunction` from the deserializer? AFAIK the Kafka-Deserializer has to return a value for each element or null? But null values should be avoided when working with streams.